go.mod

```diff
@@ -5,6 +5,7 @@ go 1.21
 toolchain go1.21.5
 
 require (
+	github.com/bxcodec/httpcache v1.0.0-beta.3
```

Comment:
I'm not sure if this is a concern or not, but just noticed this package(and assuming the gotcha package as well) have been on the same beta version for over 3 years at this point with no updates, are we ok with adopting these dependencies? if we want to accept them I could buy into that, just want to make sure we are aware of them

Reply:
Can we use ETag caching here?

Reply:
@eljohnson92 Other libraries (what i found) are archived, so i found this one a better option than writing our own.
I'm open to use something else, do you have any preference?

Reply:
@srust Maybe i'm wrong, but Linode API doesn't send ETag header, at least the for web ui. Could you please clarify your suggestion?

Reply:
@mhmxs as an alternative to this, could we perhaps use a more general purpose go caching library to store cache instead of the purpose-built http caching library? I would be open to using this if we thing it would be the best option, just want to explore some alternatives as well

Reply:
@eljohnson92 This was the simplest to implement. The alternatives i found:
- stores `string` or `[]byte`, so we have to serialize data
- stores `any` or `interface{}`, so we have to type enforce type after every get
- needs an extra layer between business logic and Llinode client
- already archived
- not designed for high concurrency
- doesn't support TTL

So i'm glad to use another library, if you have better option. This library was the best result of my researching :)

Reply:
Bummer about ETag - oh well.